core/src/main/scala/pureconfig/backend/TypesafeConfigWrapper.scala

```diff
@@ -0,0 +1,38 @@
+package pureconfig.backend
+
+import java.nio.file.Path
+
+import com.typesafe.config.{ Config, ConfigFactory }
+import pureconfig.error.{ ConfigReaderFailures, ThrowableFailure }
+
+import scala.util.control.NonFatal
+
+/**
+ * A wrapper of [[com.typesafe.config.ConfigFactory]] whose methods return `Either` instead
+ * of throwing exceptions
+ */
+object ConfigFactoryWrapper {
+
+  /** @see [[ConfigFactory.invalidateCaches()]] */
+  def invalidateCaches(): Either[ConfigReaderFailures, Unit] =
+    unsafeToEither(ConfigFactory.invalidateCaches())
+
+  /** @see [[ConfigFactory.load()]] */
+  def load(): Either[ConfigReaderFailures, Config] =
+    unsafeToEither(ConfigFactory.load())
+
+  /** @see [[ConfigFactory.parseFile()]] */
+  def parseFile(path: Path): Either[ConfigReaderFailures, Config] =
+    unsafeToEither(ConfigFactory.parseFile(path.toFile))
+
+  /** Utility methods that parse a file and then calls `ConfigFactory.load` */
+  def loadFile(path: Path): Either[ConfigReaderFailures, Config] =
+    parseFile(path).right.flatMap(rawConfig => unsafeToEither(ConfigFactory.load(rawConfig)))
+
+  private def unsafeToEither[A](f: => A): Either[ConfigReaderFailures, A] = {
+    try (Right(f)) catch {
+      case NonFatal(e) =>
+        Left(ConfigReaderFailures(ThrowableFailure(e, None)))
```

Comment:
I think we can do a little better here. If we catch [`ConfigException`](https://typesafehub.github.io/config/latest/api/com/typesafe/config/ConfigException.html), we have access to a `ConfigOrigin` which we can then use to create a `ConfigValueLocation`. This would add more context to our errors. Also, we can map some types of `ConfigException` (I see at least [`ConfigException.Parse`](https://typesafehub.github.io/config/latest/api/com/typesafe/config/ConfigException.Parse.html) and [`ConfigException.NotResolved`](https://typesafehub.github.io/config/latest/api/com/typesafe/config/ConfigException.NotResolved.html)) which can be mapped to more semantically rich failures of PureConfig, other than just a `ThrowableFailure`.